### PR TITLE
75-BE/missing-handles

### PR DESCRIPTION
**Pull request by Paurikova2**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  5  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
After handle import, there are missing some handles.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java [hunk 1]
```diff
@@ -430,6 +430,36 @@ public String getDeadSince(Context context, String handle) throws SQLException {
...
+    public Handle createHandle(Context context, String handleStr) throws SQLException, AuthorizeException {
+        // Check authorisation: Only admins may create DC types
+        if (!authorizeService.isAdmin(context)) {
+            throw new AuthorizeException(
+                    "Only administrators may modify the handle registry");
+        }
+
+        String handleId;
+        // Do we want to generate the new handleId or use entered handleStr?
+        if (!(StringUtils.isBlank(handleStr))) {
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java:443]: You can use StringUtils.isNotBlank

File: dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java [hunk 2]
```diff
@@ -430,6 +430,36 @@ public String getDeadSince(Context context, String handle) throws SQLException {
                 format(timestamptz) : null;
     }
 
+    @Override
+    public Handle createHandle(Context context, String handleStr) throws SQLException, AuthorizeException {
+        // Check authorisation: Only admins may create DC types
+        if (!authorizeService.isAdmin(context)) {
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java:436]: Only admin? Will never any user use this method during creating an Item? This is only migration purposes right?

File: dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java [hunk 3]
```diff
@@ -430,6 +430,36 @@ public String getDeadSince(Context context, String handle) throws SQLException {
...
+        } else {
+            // We generate new handleId
+            handleId = createId(context);
+        }
+
+        Handle handle = handleDAO.create(context, new Handle());
+
+        // Set handleId
+        handle.setHandle(handleId);
+
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/handle/HandleClarinServiceImpl.java:455]: why so much empty rows?

File: dspace-api/src/main/java/org/dspace/handle/service/HandleClarinService.java [hunk 4]
```diff
@@ -215,4 +215,14 @@ public void update(Context context, Handle handleObject, String newHandle,
      * @throws SQLException if database error
      */
     public String getDeadSince(Context context, String handle) throws SQLException;
+
+    /**
+     * Create handle without dspace object.
```
Comment (milanmajchrak) [inline dspace-api/src/main/java/org/dspace/handle/service/HandleClarinService.java:220]: If this is only for migration purposes please add some note that this is only for migration and for a specific case.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java [hunk 5]
```diff
@@ -101,15 +108,15 @@ public class ClarinItemImportController {
      * https://<dspace.server.url>/api/clarin/import/workspaceitem
      * }
      * </pre>
-     * @param request
-     * @return
-     * @throws AuthorizeException
-     * @throws SQLException
+     * @param request request
+     * @return workslaceitem converted to rest
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinItemImportController.java:112]: workspace not workslace

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ClarinHandleImportControllerIT.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ * Integration test to test the /api/clarin/import/handle endpoint
+ *
+ * @author Michaela Paurikova (michaela.paurikova at dataquest.sk)
+ */
+public class ClarinHandleImportControllerIT extends AbstractControllerIntegrationTest {
+    @Autowired
+    private HandleClarinService handleService;
+
+    @Test
+    public void createHandleTest() throws Exception {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/ClarinHandleImportControllerIT.java:37]: Please describe better what are you testing. Is it that handles without DSpace obejct?

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

**Review by milanmajchrak — APPROVED** [review]

**Review by mrovnanik — APPROVED** [review]

**Pull request merged**